diff --git a/apps/tuya.ai/ai_components/ai_audio/src/ai_audio_player.c b/apps/tuya.ai/ai_components/ai_audio/src/ai_audio_player.c
index 46afb49d..e8276fd9 100755
--- a/apps/tuya.ai/ai_components/ai_audio/src/ai_audio_player.c
+++ b/apps/tuya.ai/ai_components/ai_audio/src/ai_audio_player.c
@@ -382,12 +382,18 @@ OPERATE_RET ai_audio_play_tts_stream(AI_AUDIO_PLAYER_TTS_STATE_E state, AI_AUDIO
         TUYA_CALL_ERR_LOG(tuya_ai_player_start(__s_tone_player, AI_PLAYER_SRC_MEM, NULL, codec));
         ai_user_event_notify(AI_USER_EVT_TTS_START, NULL);
         break;
-    case AI_AUDIO_PLAYER_TTS_DATA:
+    case AI_AUDIO_PLAYER_TTS_DATA: {
         // PR_DEBUG("audio player -> tts stream data %d", len);        
         if (data && len > 0 && __s_tts_play_flag) {
             TUYA_CALL_ERR_LOG(tuya_ai_player_feed(__s_tone_player, (uint8_t *)data, len));
         }
         ai_user_event_notify(AI_USER_EVT_TTS_DATA, NULL);
+            AI_NOTIFY_TEXT_T tts_data;
+            tts_data.data = data;
+            tts_data.datalen = len;
+            tts_data.timeindex = 0;
+            ai_user_event_notify(AI_USER_EVT_TTS_DATA, &tts_data);
+        }
         break;
     case AI_AUDIO_PLAYER_TTS_STOP:
         PR_DEBUG("audio player -> tts stream stop");
diff --git a/apps/tuya.ai/ai_components/ai_skills/src/ai_skill.c b/apps/tuya.ai/ai_components/ai_skills/src/ai_skill.c
index b53c407c..4a54c942 100755
--- a/apps/tuya.ai/ai_components/ai_skills/src/ai_skill.c
+++ b/apps/tuya.ai/ai_components/ai_skills/src/ai_skill.c
@@ -83,9 +83,8 @@ static OPERATE_RET __ai_skills_process(cJSON *root, bool eof)
     else {
         PR_NOTICE("skill %s not handled", code);
         /* PR_NOTICE("skill content %s ", cJSON_PrintUnformatted(root)); */
-
-        ai_user_event_notify(AI_USER_EVT_SKILL, root);
     }
+    ai_user_event_notify(AI_USER_EVT_SKILL, root);
 
     return rt; 
 }
